components/capture-build-scan-url-maven-extension/src/main/java/com/gradle/ConfigureGradleEnterprise.java

```diff
@@ -0,0 +1,129 @@
+                append(errorFile, "The Gradle Enterprise server URL has not been configured in the project or on the command line.");
+            }
+        });
+
+        buildScan.buildFinished(buildResult -> {
+            String expId = System.getProperty("com.gradle.enterprise.build-validation.expId");
+            addCustomValueAndSearchLink(buildScan, "Experiment id", expId);
+            buildScan.tag(expId);
+
+            String runId = System.getProperty("com.gradle.enterprise.build-validation.runId");
+            addCustomValueAndSearchLink(buildScan, "Experiment run id", runId);
+
+            String scriptsVersion = System.getProperty("com.gradle.enterprise.build-validation.scriptsVersion");
+            buildScan.value("Build validation scripts", scriptsVersion);
+        });
+
+        buildScan.buildScanPublished(scan -> {
+            String runNum = System.getProperty("com.gradle.enterprise.build-validation.runNum");
+            URI buildScanUri = scan.getBuildScanUri();
+            String buildScanId = scan.getBuildScanId();
+            String port = buildScanUri.getPort() != -1 ? ":" + buildScanUri.getPort() : "";
+            String baseUrl = String.format("%s://%s%s", buildScanUri.getScheme(), buildScanUri.getHost(), port);
+
+            File scanFile = new File(EXPERIMENT_DIR, "build-scans.csv");
+            append(scanFile, String.format("%s,%s,%s,%s\n", runNum, baseUrl, buildScanUri, buildScanId));
+        });
+    }
+
+    private static void configureBuildScanPublishing(BuildScanApi buildScan) {
+        buildScan.publishAlways();
+        buildScan.capture(t -> t.setGoalInputFiles(true)); // also set via sys prop
+        buildScan.setUploadInBackground(false);
+    }
+
+    private static void addCustomValueAndSearchLink(BuildScanApi buildScan, String label, String value) {
+        buildScan.value(label, value);
+        if (buildScan.getServer() != null) {
+            String server = buildScan.getServer();
+            String searchParams = "search.names=" + urlEncode(label) + "&search.values=" + urlEncode(value);
+            String url = appendIfMissing(server, "/") + "scans?" + searchParams + "#selection.buildScanB=" + urlEncode("{SCAN_ID}");
+            buildScan.link(label + " build scans", url);
+        }
+    }
+
+    private static String appendIfMissing(String str, String suffix) {
+        return str.endsWith(suffix) ? str : str + suffix;
+    }
+
+    private static String urlEncode(String str) {
+        try {
+            return URLEncoder.encode(str, StandardCharsets.UTF_8.name());
+        } catch (UnsupportedEncodingException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static void append(File file, String text) {
+        try {
+            Files.write(file.toPath(), text.getBytes(), CREATE, WRITE, APPEND);
+        } catch (IOException e) {
+            throw new RuntimeException("Unable to write to file to build-scans.csv: " + e.getMessage(), e);
```

Comment:
This error message specifically references `build-scans.csv`, but we also use this function to append to `errors.txt`. Maybe something like this would make the message generic:

```suggestion
            throw new RuntimeException(String.format("Unable to write to %s: %s", file.getName(), e.getMessage(), e);
```

Reply:
An oversight. I'll fix it.